Assets/Scripts/HealthController.cs

```diff
@@ -0,0 +1,34 @@
+using UnityEngine;
+
+public class HealthController : MonoBehaviour
+{
+    [SerializeField]
+    private int maxhHealth = 100;
+    public int MaxHealth{get => maxhHealth;}
+
+    private int currentHealth = 0;
+
+    public int CurrentHealth { get => currentHealth; }
+
+    public delegate void damageEvent(HealthController healthController, int damage);
+    public damageEvent OnDamageTaken;
+
+    private void Awake()
+    {
+        foreach(var box in GetComponentsInChildren<HitboxController>())
+        {
+            box.health = this;
```

Comment:
Yeah we'll deal with the structure of this later. Not sure if this is ideal :)))))